modules/dynamic_creds/main.tf

```diff
@@ -0,0 +1,51 @@
+data "google_project" "project" {
```

Comment:
🤔  I wonder whether we should move this into https://github.com/humanitec-architecture/resource-packs-gcp as a standalone definition people could install and then re-use this here as well.

Reply:
Hmm... I got your point, however this module does not include any resource definitions. Additionally, resource pack resources operate differently. They are provisioned via Humanitec only if there are workloads that depend on them. In contrast, this module creates a cloud account independently of Humanitec

Reply:
👍 , I'm mainly concerned that we need similar logic now in https://github.com/humanitec-architecture/resource-packs-gcp/blob/main/examples/dns/main.tf#L1-L29 (and other places where accounts might be used) and that customers don't have an easy way create a linked service-account.

In any case, could also be extracted as a follow-up.